### ISSUE TITLE
False positives on react/jsx-props-no-multi-spaces with generic component

### ISSUE THREAD

**Issue by luixo** [yannickcr/eslint-plugin-react#2181]
Setup: `eslint` 5.14.1 with `@typescript-eslint/eslint-plugin` 1.4.0
Minimal case:
```ts
<Foo<T> bar />
```
Error:
`Expected only one space between "Foo" and "bar"`

**Comment by ljharb** [comment]
It's very surprising that TS allows that; `Foo<T>` looks like a type, but the first item in a jsx element has to be a *value*.

**Comment by G-Rath** [comment]
> It's very surprising that TS allows that

That's how you do [generic JSX components in TypeScript](https://github.com/Microsoft/TypeScript/issues/6395).

**Comment by ljharb** [comment]
I'm still surprised they chose to deviate from the jsx spec for that.

**Comment by G-Rath** [comment]
But is it actually deviating from the spec, since it's never actually in the jsx when it gets compiled?

Like, I think its fuzzy on both sides - I can't think of a cleaner way they could have done it 😕

**Comment by ljharb** [comment]
Yes, it is. I'd say `<Foo: Foo<T> />` would be much cleaner, and would stick to the "stuff after `:` is types" space.

That ship has obviously sailed, though.

**Comment by G-Rath** [comment]
I've not got the jsx spec in front of me, but why is it deviating from the spec to have a generic type straight after the component name, but not a colon?

**Comment by ljharb** [comment]
I suppose it’s more than in typescript itself, generics syntax only appears in type space, and in this case, it’s appearing in value space - so it seems inconsistent with typescript too.

**Comment by G-Rath** [comment]
Fair enough. What are the chances of getting a fix? Even if it has to be behind a `typescript: true` style option?

**Comment by ljharb** [comment]
I don't think it has to be behind an option; this change would be removing a false error only for typescript users.

I think a PR would be fine.
